Source/WebCore/html/canvas/CanvasRenderingContext2DBase.h

```diff
@@ -247,11 +248,14 @@ class CanvasRenderingContext2DBase : public CanvasRenderingContext, public Canva
     using Direction = CanvasDirection;
     void setDirection(Direction);
 
-    class FontProxy final : public FontSelectorClient {
-    public:
-        FontProxy() = default;
+    class FontProxy final : public RefCounted<FontProxy>, public FontSelectorClient {
+
+        public:
+        static RefPtr<FontProxy> create() { return adoptRef(*new FontProxy()); }
         virtual ~FontProxy();
         FontProxy(const FontProxy&);
```

Comment:
Should not be public. Also, it is fishy for a ref-counted class to have a copy constructor.